### ISSUE TITLE
[Bug]  Error "the path cannot be traversed because it contains an untrusted mount point" when executing apps requiring Administrator privileges with Windows 11 Insiders preview 25236

### ISSUE THREAD

**Issue by okibcn** [ScoopInstaller/Scoop#5238]
#### Current Behavior
After upgrading to Windows 11 Insider Preview 25236.1000, any app requiring administrative privilege is not working using the links in the Start menu. Tested with ```wiztree``` and ```iobitdriverbooster```. It returns the error "the path cannot be traversed because it contains an untrusted mount point"

Browsing to the app folder through the ```SCOOP_FOLDER/apps/APPNAME/current/``` in explorer.exe or executing it by 🪟+R sortcut returns the error.

However, it is important to mention that browsing to the app folder through the ```SCOOP_FOLDER/apps/APPMANE/APPVERSION/``` in explorer.exe or executing it by 🪟+R shortcut doesn't return error and works perfectly as before. This indicates that the folder shortcut ```current``` could be the source of the lack of trust in that execution path.

#### Expected Behavior
Normal execution after asking for elevated privileges.

#### Additional context/output

![image](https://user-images.githubusercontent.com/22417711/199625724-7f7a3fa1-986f-424d-b82e-bd5e99e928d0.png)

#### Possible Solution
A simple workaround could be to stop using the ```current``` folder shortcut and update the app shortcuts to the desired version folder. That works as a workaround so far.

### System details
```powershell
Name                           Value
----                           -----
PSVersion                      7.2.7
PSEdition                      Core
GitCommitId                    7.2.7
OS                             Microsoft Windows 10.0.25236
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```
#### Scoop Configuration
```json
{
  "aria2-warning-enabled": false,
  "last_update": "2022-11-02T15:14:58.1081954-07:00",
  "scoop_branch": "master",
  "scoop_repo": "https://github.com/ScoopInstaller/Scoop"
}
```

**Comment by okibcn** [comment]
Note that using the ```current``` folder shortcut has a history of conflicting with outer security features of Windows such as the firewall referred to in issue #5234. 
The use of the ```current``` folder makes it easy to have a default folder while maintaining the other versions, but it is now a problem affecting more than firewall-related permissions, and with the latest Windows 11 dev preview release, it will collide with all those apps requiring administrative privileges.
So far I am modifying the shortcuts to point to the desired version. This could be the way forward, removing the ```current``` folder link and modifying the shortcuts to the apps instead of the current folder shortcut.

**Comment by Still34** [comment]
Can confirm, reproducible on 25236.

**Comment by niheaven** [comment]
We are planning to rewrite the app structure but this may take some time. Thank you all for the bug reporting and potential solutions.

**Comment by rashil2000** [comment]
Hi, I'm unable to reproduce this. I'm on build 25236. Could anyone tell me a minimal reproducible example?

**Comment by okibcn** [comment]
@rashil2000 find here detailed steps to reproduce.

1. Ensure you are in Windows 11 insiders dev build 25236 (the latest at this time) with the default security settings.
2. install scoop
```powershell
Set-ExecutionPolicy RemoteSigned -scope CurrentUser
iwr -useb get.scoop.sh | iex
```
3. Add the 'extras' bucket
```powershell
scoop bucket add extras
```
4. Install an app that requires elevated rights such as wiztree from the extras bucket
```powershell
scoop install wiztree
```
5. Run Wiztree from start menu or by typing ```wiztree``` in Powershell. Wiztree is not in administrator mode at startup, go to the 'options' in the menu bar and select 'Always Run as Administrator' and close the app.
6. The problem is now reproducible. Just try to execute Wiztree again from powershell or from start menu and it wont start. If you right click on the Wiztree icon through Windows Start button and select 'Run As Administrator' the error dialog is visible:
![image](https://user-images.githubusercontent.com/22417711/201726380-7c6295fe-4281-4b20-8033-f51c0ce3f23f.png)


To track down the problem open a windows powershell (Not Administrator) and navigate to the wiztree app folder. Just see the different behavior using these two scripts.
FAILING USING CURRENT FOLDER:
```powershell
cd
cd .\scoop\apps\wiztree\current
.\wiztree.exe
```
WORKING USING VERSION FOLDER (note 4.11 is the current wiztree version, it can differ in the future):
```powershell
cd
cd .\scoop\apps\wiztree\4.11
.\wiztree.exe
```

So, the problem seems to be directly related to the use of a linked directory conflicting with the modified default security settings after the latest windows insiders dev update 25236, as it was working fine prior to this update.

**Comment by okibcn** [comment]
@xiiok 
I did something similar, but I also included the elevated prompt request using sudo app, not requiring opening an elevated prompt for that. Curiously my function was named ```scoopfix```. And I run it manually every time any of the scoop apps are updated, or I install any new one. It requires the ```sudo``` app, something I use quite frequently. You can get it with ```scoop install sudo```, and this is my function as it looks now, based on yours:
```powershell
function scoopfix
{
  function scoopfix_ ($app) {
    $app ? ($path="$env:USERPROFILE\scoop\apps\$app") : ($path="$env:USERPROFILE\scoop\apps") > $null
    echo "Trusting junctions:"
    ls $path current -recurse -depth 1 -at reparse | %{
        echo $_.target
        $acl=acl $_
        $target=$_.target
        attrib $_ -R /L
        rm $_
        $current=ni $_ -item junction -value $target
        set-acl -path $current -acl $acl
        attrib $current +R /L
    }
  }
  sudo pwsh -c scoopfix_
}
```

As you say, it is a quick and dirty solution, but it works while using a standard session.

**Comment by dozius** [comment]
Worth mentioning that it's not just the `current` folder that is an issue. If the app has any persisted junctions these will also be inaccessible and cause all kinds of issues. The workarounds above will let you launch the app but you will also need to go in and fix all the persisted junctions for the app in order for it to function correctly.

**Comment by okibcn** [comment]
@dozius I have several apps requiring elevated rights and have junctions in the app folder targeting its folder under the persist directory. I never had that problem. What app is giving you that kind of problem?

In any case, the solution is quite easy, just remove the ```-depth 1``` option and that's all you need to do. So, the solution would be adding this to the PowerShell profile by editing it with ```notepad $PROFILE```:
```pwsh
function scoopfix_ ($app) {
  $app ? ($path="$env:USERPROFILE\scoop\apps\$app") : ($path="$env:USERPROFILE\scoop\apps") > $null
  echo "Trusting junctions:"
  ls $path current -recurse -at reparse | %{
      Write-Host -n "$_ -> " 
      Split-Path $_.target -leaf
      $acl=acl $_
      $target=$_.target
      attrib $_ -R /L
      rm $_
      $current=ni $_ -item junction -value $target
      set-acl -path $current -acl $acl
      attrib $current +R /L
  }
}

function scoopfix ($app)
{
  sudo pwsh -c "scoopfix_ $app"
} 
```
Just close and reopen the PowerShell terminal, or execute ```& $PROFILE``` to have that function available. It requires the sudo command that you can install with ```scoop install sudo``` and the usage is as easy as calling ```scoopfix``` for fixing all the junctions, or ```scoop APP_NAME`` to fix only the junctions related to that single app.

**Comment by dozius** [comment]
Yes, I already fixed it on my end but thanks for the detailed write up.

The specific app was OBS. It would throw a few errors related to the config path and then crash. Fixing the persisted junction fixed the problem. I launch the application from a streamdeck normally, so maybe this factors into it somehow.

In any case, the junction for persisted data was an issue. After I fixed it everything was working as intended again.

**Comment by seritools** [comment]
In case someone runs into this issue now and stumbles upon this github issue:

The KB5074105 preview update seems to break junctions over ssh (tested with VSCode remote), even if marked as trusted by the script above. Only uninstalling the update helped so far.

Feedback hub link: https://aka.ms/AAzhjzy
